### DISCUSSION TITLE
Changing menu items based on Mobile vs Desktop

### DISCUSSION THREAD

**Discussion by Babilinski** [facebook/docusaurus#5874] [Q&A]
I am running Docusaurus without an index page and created header links (menu items in the top navigation bar) for the different sections in my website. This works fine on desktop. However, navigating becomes difficult when using a tablet or phone, because the user needs to select "Back to main menu" to view the header links.

Is there a way to change the menu items in the config file depending on what device it is being viewed on? 

I think header links should remain un-nested at the top of the navigation bar when viewing it on mobile, this would be more consistent with how the header links work on desktop.

**Comment by slorber** [comment]
> This works fine on desktop. However, navigating becomes difficult when using a tablet or phone, because the user needs to select "Back to main menu" to view the header links.

This is the docs mobile sidebar navigation panel that we really need, otherwise user wouldn't be able to navigate. It's not clear what alternative UX you are suggestion exactly, and how sidebar navigation would work with this new UX.

> Is there a way to change the menu items in the config file depending on what device it is being viewed on?

We'll likely implement something like this in https://github.com/facebook/docusaurus/issues/4389

> I think header links should remain un-nested at the top of the navigation bar when viewing it on mobile, this would be more consistent with how the header links work on desktop.

On Desktop, header links and docs sidebar links are not displayed in the same place, and we do exactly the same on mobile, it's just using a different layout. Please make a concrete UX proposal with mock-ups, taking into consideration the whole picture and we'll consider doing some changes.

**Comment by Babilinski** [comment]
Thank you for your suggestions, I'll create UX mockups of this the issue I'm describing.

**Comment by jdarpinian** [comment]
I got rid of the "Back to main menu" thing on my site, which is just a blog, by wrap-swizzling BlogSidebar/Mobile and returning null instead of the original component. Now the hamburger menu just shows the header links like you and I wanted.
